src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java

```diff
@@ -483,6 +484,19 @@ private boolean matches(Offer offer, Request request) {
     }
   }
 
+  private boolean buildsInQueue(hudson.model.Queue queue) {
+   hudson.model.Queue.Item[] items =  queue.getItems();
```

Comment:
I'd prefer something like getApproximateItemsQuickly() to prevent thread contention.